server/portal/libs/agave/utils.py

```diff
@@ -158,7 +158,7 @@ def text_preview(url):
     """
     try:
         resp = requests.get(url)
-        if (resp.encoding is not None and resp.encoding.lower() == 'utf-8'):
+        if (resp.content or (resp.encoding is not None and resp.encoding.lower() == 'utf-8')):
```

Comment:
Is this change related to the 500 error? If someone tries to preview a binary/executable I think they'll get a bunch of gibberish.

Reply:
Unrelated but got accidentally folded in. However I do think it's worth thinking about. I don't think the user would get any gibberish since the `content.encode('ascii', 'strict')` would raise an error if it can't decode the text. I did confirm this with a `tapisjob.sh` file. These files don't seem to have anything in `resp.encoding`, but they are previewable text files